src/H5Odtype.c

```diff
@@ -1644,6 +1710,13 @@ H5O__dtype_size(const H5F_t *f, const void *_mesg)
             ret_value += H5O__dtype_size(f, dt->shared->parent);
             break;
 
+        case H5T_COMPLEX:
+            if (dt->shared->u.cplx.homogeneous)
+                ret_value += H5O__dtype_size(f, dt->shared->parent);
+            else
+                ret_value = 0;
```

Comment:
'0' seems unhelpful.  Throw an error instead?

Reply:
The function currently returns a `size_t` (0 for failure), though it could be updated.

Reply:
Ah, OK, didn't expand upward.  :-)